.github/workflows/build.yaml

```diff
@@ -39,50 +72,170 @@ jobs:
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v3
+      - name: Identify week number
+        shell: bash
+        run: |
+          date +'%V' > week-number.txt
+          date +'week-number=%V' >> $GITHUB_ENV
       - uses: actions/setup-python@v4
+        id: setup-python
         with:
           python-version: ${{ matrix.python-version }}
+          cache: "pip"
+          cache-dependency-path: |
+            .github/workflows/build.yaml
+            setup.cfg
+            setup.py
+            tox.ini
+            week-number.txt
+      - name: Restore cache
+        id: restore-cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            .tox
+            .venv
+          key: >
+            test
+            week=${{ env.week-number }}
+            os=${{ matrix.os }}
+            python=${{ steps.setup-python.outputs.python-version }}
```

Comment:
This applies to all jobs using this as a cache key, not just this one.

I think I saw another build using the python location from the env.
Two (conflicting) thoughts:
- I like this better, it reads more cleanly
- will this potentially result in corrupt/bad caches when github updates point releases (e.g. a 3.11.0 virtualenv running with a 3.11.1 interpreter)?

Reply:
In another thread/context it came out that this version includes the patch version number, not just the input/requested version number. So I think I have no concern here. 👍